library/core/src/slice/mod.rs

```diff
@@ -3994,6 +3994,76 @@ impl<T> [T] {
     }
 }
 
+#[cfg(not(bootstrap))]
+impl<T, const N: usize> [[T; N]] {
+    /// Takes a `&[[T; N]]`, and flattens it to a `&[T]`.
+    ///
```

Comment:
suggestion: include another section in here for the panic, something like
```suggestion
    ///
    /// # Panics
    ///
    /// If the length of the resulting slice would overflow a `usize`.
    ///
    /// This is only possible when flattening a slice of arrays of
    /// zero-sized types, and thus tends to be irrelevant in practice.
    ///
    /// If `size_of::<T>() > 0`, this will never panic.
    ///
```

Reply:
Ah good idea